### DISCUSSION TITLE
Difference between the two EXE files.....or, what is "pcbasicw.exe" for?

### DISCUSSION THREAD

**Discussion by matamobo** [robhagemans/pcbasic#227] [Q&A] [answered]
Hi, love this program and the fond memories it brings back for me. Kudos to you and all the developers for creating such a masterpiece. I cut my teeth on M$Basic on CP/M, moving on to the IBM PC's and XT's with PC/GWbasic and beyond.  Sometimes simple is the best solution. My only question is, what is the difference between pcbasic.exe and pcbasicw.exe in the zip package for windows (other than the "w" version seems to open a command box before opening the actual basic interpreter window). I apologize if this is in the documentation, but I was unable to find it after looking for quite some time. Thanks for your time.

**Comment by Marrin** [comment, marked as answer]
That's the difference.  It's a Python + Windows thing.  In Windows there's a difference between GUI apps and console apps.  Python has standard input and output channels and the EXE that opens the console in addition to the graphical window are connected to those channels.  So you can see things like crashs there or choose another interface than the graphical one.
